### PR TITLE
Skip __main__.py

### LINKED ISSUE #229: Tries to import `__main__.py`
`__main__.py` doesn’t need a `if __name__ == '__main__'`, since it just exists to be a CLI

Therefore doctestplus shouldn‘t try to import it.

### PR DESCRIPTION
**Pull request by flying-sheep**
Fixes #229

### CODE REVIEW

File: pytest_doctestplus/plugin.py [hunk 1]
```diff
@@ -228,7 +228,7 @@ def collect(self):
                 fspath = self.fspath
                 filepath = self.fspath.basename
 
-            if filepath == "setup.py":
+            if filepath in {"setup.py", "__main__.py"}:
```
Comment (pllim) [inline pytest_doctestplus/plugin.py:231]:
```suggestion
            if filepath in ("setup.py", "__main__.py"):
```

**Comment by pllim** [comment]
Thanks! Please also add a change log entry to https://github.com/scientific-python/pytest-doctestplus/blob/main/CHANGES.rst

**Comment by pllim** [comment]
The diff LGTM! I would use the "Squash and Merge" button but it is currently disabled by this repo.

@bsipocz , should we enable "Squash and Merge"?

**Review by pllim — APPROVED** [review]
Thanks!

**Comment by flying-sheep** [comment]
Sure :smile: 

I don’t quite get your “Nitpick” change though: I preserved the number of newlines between the last changelog entry and the next header intentionally :wink:

**Comment by pllim** [comment]
> I don’t quite get your “Nitpick” change though

Yeah, sorry. Strictly speaking, preserving the existing number of blank lines is more correct, though I do not really care if it is two or one, personally.

**Comment by bsipocz** [comment]
I don't like squash and merge and would not like to use it here, instead prefer to use the release scripts to do changelog consistency and nitpicks prior release, rather than in the PRs.

**Comment by flying-sheep** [comment]
Is there anything left to do for me or is this good to go?

**Review by bsipocz — APPROVED** [review]
I've removed the nitpick and revert commit, and will go ahead with merging. Thanks @flying-sheep!

**Pull request merged**